src/GraphQL/PersistedDocuments/PersistedDocumentOptions.cs

```diff
@@ -0,0 +1,45 @@
+namespace GraphQL.PersistedDocuments;
+
+/// <summary>
+/// A set of options for <see cref="PersistedDocumentHandler"/>.
+/// </summary>
+public class PersistedDocumentOptions
+{
+    private const string ERROR_NO_REQUESTSERVICES = $"{nameof(ExecutionOptions)}.{nameof(ExecutionOptions.RequestServices)} or {nameof(PersistedDocumentOptions)}.{nameof(GetQueryDelegate)} must be set.";
+    private const string ERROR_NO_LOADER = $"{nameof(IPersistedDocumentLoader)} must be registered with your DI provider or {nameof(PersistedDocumentOptions)}.{nameof(GetQueryDelegate)} must be set.";
+
+    /// <summary>
+    /// Indicates if requests that are not persisted document requests should be allowed to be executed.
+    /// </summary>
+    public bool AllowNonpersistedDocuments { get; set; }
```

Comment:
I believe using this feature for performance is a more common use case than for the allow list. So, I believe this should be `true` by default. Or renamed to `AllowOnlyPersistedDocuments`.

Reply:
See comments above; APQ support is not possible with the current state of the draft RFC appendix.  As such, I believe the primary purpose of this PR is for trusted document support.